src/core/server/elasticsearch/elasticsearch_service.mock.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { BehaviorSubject } from 'rxjs';
+import { ClusterClient } from './cluster_client';
+import { ElasticsearchConfig } from './elasticsearch_config';
+import { ElasticsearchService, ElasticsearchServiceStart } from './elasticsearch_service';
+
+const startContract: ElasticsearchServiceStart = {
```

Comment:
**Note**: right now it is declared as a singleton, which is opposed to `elasticsearchServiceMock` factory approach.

Reply:
Any reason why we wouldn't just do the `createStartContract` in this PR? Is the change bigger than simply updating the references?

I agree that we can probably rely on the correctness of the arguments being enforced by the type system.

Reply:
If we agree on ` arguments being enforced by the type system.` then I will add the factory in the current PR

Reply:
Works for me

Reply:
@epixa updated

Reply:
I still see a singleton `startContract` being exported rather than a `createStartContract` factory. Am I missing something?

Reply:
I didn't push the last commit.  the classical problem when working on multiple branches. pushed